### PR TITLE
Story #645

### PR DESCRIPTION
**Pull request by michaelkleinhenz**
Implements the planner npm lib with a runtime environment and an extra external library project for shared components and services. Closes story #645.

### CODE REVIEW

**Comment by michaelkleinhenz** [comment]
I know this is a big big PR, but in that case, two whole new sub projects were added and it just did not make sense to seperate them out.

**Comment by michaelkleinhenz** [comment]
[test]

File: runtime/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Minimal Runtime for the Planner.
+
+To use this with the Planner library:
+
+0. Get dependencies for Planner: `npm install`
+1. Build the Planner library: `gulp build:library`
+2. Go to the library build directory and link to npm cache: `npm link`
```
Comment (kwk) [inline runtime/README.md]: Please explain what "go to the library build directory" means. On point 3. you also explain `cd runtime`.

Comment (michaelkleinhenz) [inline runtime/README.md]: I changed it to something a bit more clear.

**Pull request merged**